# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/ArrayBufferObject.h
# Commit: e39ae7aad8d9
# Full Hash: e39ae7aad8d93497b7a7e6ccf4ee675b4631c28e
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2023-09-21 03:21:32
# Regressor Bug: 1690111
# File Overlap Count: 2
# Description:
#   Bug 1690111 - Implement JS::EnsureNonInlineArrayBufferOrView. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D176596
# ==============================================================================

diff -r 603e8067e68f -r e39ae7aad8d9 js/src/vm/ArrayBufferObject.h
--- a/js/src/vm/ArrayBufferObject.h	Wed Sep 20 09:41:59 2023 +0000
+++ b/js/src/vm/ArrayBufferObject.h	Wed Sep 20 09:42:00 2023 +0000
@@ -416,6 +416,8 @@
     return true;
   }
 
+  static bool ensureNonInline(JSContext* cx, Handle<ArrayBufferObject*> buffer);
+
   // Detach this buffer from its original memory.  (This necessarily makes
   // views of this buffer unusable for modifying that original memory.)
   static void detach(JSContext* cx, Handle<ArrayBufferObject*> buffer);